src/control/lib/hardware/pci.go

```diff
@@ -141,8 +139,15 @@ func (pa *PCIAddress) LessThan(other *PCIAddress) bool {
 		return false
 	}
 
-	return pa.VMDAddr.LessThan(other.VMDAddr) ||
-		pa.Domain < other.Domain ||
+	// If VMD backing device address, return early on domain comparison if VMD domains are not
+	// equal. If equal, proceed to sort on backing device address BDF.
+	if pa.VMDAddr != nil && other.VMDAddr != nil {
+		if !pa.VMDAddr.Equals(other.VMDAddr) {
+			return pa.VMDAddr.LessThan(other.VMDAddr)
+		}
+	}
```

Comment:
What happens if one or the other is nil? I think you probably need some logic for these cases, something like:
```go
switch {
        case pa != nil && other == nil:
                return false
        case pa == nil && other != nil:
                return true
        case pa != nil && other != nil:
                if !pa.VMDAddr.Equals(other.VMDAddr) {
                        return pa.VMDAddr.LessThan(other.VMDAddr)
                }
        }
}
```

Reply:
Ah, forgot that there's a check at the top of the method. In hindsight, I'm not 100% sure that the logic is correct for all cases without the switch statement, but if it works reliably for the current cases I'm fine with perhaps expanding the test coverage and cases in a follow-up.

Reply:
If either address is not a VMD backing device address then the standard comparison between domain and BDF is appropriate. In practice there will never be a case where backing devices are being compared with "normal" PCI addresses anyway.